lib/parlour/rbi_generator/parameter.rb

```diff
@@ -42,7 +42,7 @@ def initialize(name, type: nil, default: nil)
 
         @kind = :keyword if kind == :normal && name.end_with?(':')
 
-        @type = type
+        @type = type || 'T.untyped'
```

Comment:
When `type` is `nil` it would show up as `T.untyped` in the generated sig. However, when comparing `Parameter`s during the merge step nil and T.untyped would be treated differently. 

Moving the `|| 'T.untyped'` up here fixes that equality check and simplifies the class since we don't need to handle this possibly being nil.